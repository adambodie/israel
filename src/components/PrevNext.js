import React from 'react'
import { Link } from 'react-router-dom'

const PrevNext = props => (
	<div className='prev-next'>
		<Link to={props.prev} className='custom-button'>Prev</Link>
		<Link to={props.next} className='custom-button'>Next</Link>
	</div>
)

export default PrevNext
