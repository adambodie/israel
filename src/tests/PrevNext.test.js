import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router';
import { Link } from 'react-router-dom';

import PrevNext from '../components/PrevNext';
import {
  render,
  cleanup,
  fireEvent
} from 'react-testing-library'

afterEach(cleanup)

it('renders without crashing', () => {
	const div = document.createElement('div');
	ReactDOM.render(<MemoryRouter><PrevNext prev={'/prev'} next={'/next'}/></MemoryRouter>, div);
	ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly', () => {
	const tree = renderer.create(
    <MemoryRouter>
      <PrevNext prev={'/prev'} next={'/next'} />
    </MemoryRouter>
  ).toJSON();
	expect(tree).toMatchSnapshot();
});