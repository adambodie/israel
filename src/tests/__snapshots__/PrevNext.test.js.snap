// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="prev-next"
>
  <a
    className="custom-button"
    href="/prev"
    onClick={[Function]}
  >
    Prev
  </a>
  <a
    className="custom-button"
    href="/next"
    onClick={[Function]}
  >
    Next
  </a>
</div>
`;
